wandb/apis/public.py

```diff
@@ -2494,7 +2493,9 @@ def __init__(self, client, entity, project, name, attrs=None):
             and a["artifactCollectionName"] == self._sequence_name
         ]
         self._manifest = None
+        self._manifest_json = None
```

Comment:
Hmm, I don't see why we need to store both _manifest_json, and _manifest. It seems safer to only store one of them, otherwise programmers who modify this class need to consider both all the time.

Also this class probably shouldn't have a _manifest_json attribute, because it exposes the details of the json layout here, instead of via a class that owns those details (the ArtifactManifest class).

Reply:
Fair point. It was mostly a performance optimization to avoid re downloading the manifest when it is loaded for the first time (since we get it in this payload). However, I removed per your advice, while still preserving the performance benefits. Now the manifest is just instituted at construction time. Thanks for pointing this out.